### PR TITLE
MODORDERS-247 Make contributor-name-type configurable

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODORDERS-247](https://issues.folio.org/browse/MODORDERS-247) Make contributor-name-type configurable

## Approach
- updated logic to use `contributor.contributorNameTypeId` instead of value from mod-config
- adding verification that `contributorNameTypeId` exist in inventory-storage

#### TODOS and Open Questions

- [ ] Merge with folio-org/mod-orders-storage#137
- [x] Point to acq-models master commit once folio-org/acq-models#145 is merged

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [x] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [x] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [x] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -237,7 +237,8 @@
           "pathPattern": "/orders/pieces",
           "permissionsRequired": ["orders.pieces.item.post"],
           "modulePermissions": [
-            "orders-storage.pieces.item.post"
+            "orders-storage.pieces.item.post",
+            "orders-storage.pieces.item.get"
```
Comment (piotr-kalashuk) [inline descriptors/ModuleDescriptor-template.json]: Probably this one was supposed to go to `"PUT"` section?

File: src/main/java/org/folio/rest/impl/InventoryHelper.java [hunk 2]
```diff
@@ -368,14 +369,55 @@ private String buildLookupEndpoint(String type, Object... params) {
...
+    List<String> ids = contributors.stream()
+      .map(Contributor::getContributorNameTypeId)
+      .distinct()
+      .collect(toList());
+
+    return getContributorNameTypes(ids)
+      .thenAccept(contributorNameTypes -> {
+
+        List<String> retrievedIds = contributorNameTypes.stream()
+          .map(o -> o.getString(ID).toLowerCase())
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/InventoryHelper.java]: is `toLowerCase` really required?

File: src/main/java/org/folio/rest/impl/InventoryHelper.java [hunk 3]
```diff
@@ -423,11 +463,11 @@ private JsonObject buildInstanceRecordJsonObject(CompositePoLine compPOL, JsonOb
       instance.put(INSTANCE_PUBLICATION, new JsonArray(singletonList(publication)));
     }
 
-    if(compPOL.getContributors() != null && !compPOL.getContributors().isEmpty()) {
+    if(isNotEmpty(compPOL.getContributors())) {
       List<JsonObject> contributors = compPOL.getContributors().stream().map(compPolContributor -> {
         JsonObject invContributor = new JsonObject();
         // According MODORDERS-204 default value for all the contributors is "Personal name".
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/InventoryHelper.java]: This comment can be removed

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (pautau-s) [reply descriptors/ModuleDescriptor-template.json]: Oh, I missed. Fixed.

File: src/main/java/org/folio/rest/impl/InventoryHelper.java (hunk 2 above)
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: I wanted to guarantee that `contributorNameTypeIds` from the inventory-storage will be in the same case as from the request. I forgot to lowercase `contributorNameTypeIds` from the request.

File: src/main/java/org/folio/rest/impl/InventoryHelper.java (hunk 3 above)
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: Done

**Pull request merged**